vernac/vernacinterp.mli

```diff
@@ -8,7 +8,11 @@
 (*         *     (see LICENSE file for the text of the license)         *)
 (************************************************************************)
 
-(** The main interpretation function of vernacular expressions *)
+(** The main interpretation function of vernacular expressions.
+
+    Caveat: we don't correctly handle proof modes in the case of nested proofs,
+    in particular ending the inner one does not restore the proof mode of the
+    outern one if it is not the standard one.  *)
```

Comment:
Right, I saw this. What I mean to say was: Can we handle this correctly?

Reply:
Yes, the plan is sketched here and there, but I believe it is not worth it. Nested proofs have very little interest, and nesting different proof modes zero.

One would need to make the classifier (not the STM one, the other one) aware of proof mode changes, to begin with.
Then vernacinterp coul carry a stack of proof modes, and pop as proofs end.

Reply:
Now that I think more about it, this problem disappears when parsing and execution are separate. This PR move the proof mode "state" next to the parsing one. That state will be managed, correctly, by the synterp phase.

So this partially incorrect change will disappear soonish.